node/node.go

```diff
@@ -0,0 +1,106 @@
+package node
+
+import (
+	"context"
+	"fmt"
+	abci "github.com/lazyledger/lazyledger-core/abci/types"
+	"github.com/lazyledger/lazyledger-core/libs/log"
+	"github.com/lazyledger/lazyledger-core/libs/service"
+	tmnode "github.com/lazyledger/lazyledger-core/node"
+	"github.com/lazyledger/lazyledger-core/proxy"
+	"github.com/lazyledger/lazyledger-core/types"
+)
+
+var _ tmnode.NodeInterface = &Node{}
```

Comment:
This line might need to change, if we address https://github.com/lazyledger/lazyledger-core/pull/166#discussion_r583498992 as the interface would live in the SDK. Hmm, actually: this might not be the smartest decisions either. Because then optimint would import the SDK and the SDK would import optimint. 

Can't these assertions live in the SDK either?

Reply:
Either way, let's not overthink this now. We can also merge all 3 PRs as is and figure out the proper abstractions as we iterate over the code.

Reply:
This assertion is only to ensure that we're not braking anything during development.
As we're moving to scaffolding based approach and try to avoid all changes in tendermint/lazyledger-core, we can get rid of this interface at all. or keep it here just to reflect the requirements from SDK.